packages/contracts-bedrock/contracts/libraries/Encoding.sol

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import { Hashing } from "./Hashing.sol";
+import { RLPWriter } from "./rlp/RLPWriter.sol";
+
+/**
+ * @title Encoding
+ * @notice Encoding handles Optimism's various different encoding schemes.
+ */
+library Encoding {
+    /**
+     * Generates the correct cross domain calldata for a message.
+     * @param _target Target contract address.
+     * @param _sender Message sender address.
+     * @param _message Message to send to the target.
+     * @param _messageNonce Nonce for the provided message.
+     * @return ABI encoded cross domain calldata.
+     */
+    function encodeXDomainCalldata(
+        address _target,
+        address _sender,
+        bytes memory _message,
+        uint256 _messageNonce
+    ) internal pure returns (bytes memory) {
+        return
+            abi.encodeWithSignature(
+                "relayMessage(address,address,bytes,uint256)",
+                _target,
+                _sender,
+                _message,
+                _messageNonce
+            );
+    }
+
+    /**
+     * @notice RLP encode a deposit transaction
+     * This only works for user deposits, not system deposits
+     * TODO: better name + rearrange the input param ordering?
```

Comment:
I think this is a good thing to do.
Would be nice to do it here rather than introducing a new TODO.

Reply:
gotcha